packages/docusaurus-theme-classic/src/theme/NavbarItem/NavbarNavLink.tsx

```diff
@@ -54,11 +55,21 @@ export default function NavbarNavLink({
                 }
               : null),
           })}
-      {...props}>
-      {label}
-      {isExternalLink && (
-        <IconExternalLink {...(isDropdownLink && {width: 12, height: 12})} />
-      )}
-    </Link>
+      {...props}
+      {...(html
+        ? {dangerouslySetInnerHTML: {__html: html}}
+        : {
+            children: (
+              <>
+                {label}
+                {isExternalLink && (
+                  <IconExternalLink
+                    {...(isDropdownLink && {width: 12, height: 12})}
+                  />
+                )}
+              </>
+            ),
+          })}
+    />
```

Comment:
Non-blocking but this looks a bit messy & too deeply nested for me. If you can find a better way to structure this (I don't have much idea about how) before merge feel free to refactor it

Reply:
I have that feeling too, but can't think of a better solution.

Reply:
refactored a bit 👍